IpmiFeaturePkg/Library/MockIpmi/IpmiBaseLibMock.c

```diff
@@ -0,0 +1,86 @@
+/** @file
+  Implements the IPMI base lib for the mock BMC.
+
+  Copyright (c) Microsoft Corporation
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include "MockIpmi.h"
+#include <Library/IpmiBaseLib.h>
+
+#define MOCK_BASE_IPMI_BUFFER_SIZE  (MAX_UINT8)
+
+STATIC UINT8  RequestBuffer[MOCK_BASE_IPMI_BUFFER_SIZE];
+STATIC UINT8  ResponseBuffer[MOCK_BASE_IPMI_BUFFER_SIZE];
+
+/**
+  Initialize the global varible with the pointer of IpmiTransport Protocol.
+
+  @retval EFI_SUCCESS - Always return success
+
+**/
+EFI_STATUS
+InitializeIpmiBase (
+  VOID
+  )
+{
+  return EFI_SUCCESS;
+}
+
+/**
+  Routine to send commands to BMC.
+
+  @param NetFunction       - Net function of the command
+  @param Command           - IPMI Command
+  @param CommandData       - Command Data
+  @param CommandDataSize   - Size of CommandData
+  @param ResponseData      - Response Data
+  @param ResponseDataSize  - Response Data Size
+
+  @retval EFI_NOT_AVAILABLE_YET - IpmiTransport Protocol is not installed yet
+
+**/
+EFI_STATUS
+IpmiSubmitCommand (
+  IN UINT8    NetFunction,
+  IN UINT8    Command,
+  IN UINT8    *CommandData,
+  IN UINT32   CommandDataSize,
+  OUT UINT8   *ResponseData,
+  OUT UINT32  *ResponseDataSize
+  )
+{
+  IPMI_COMMAND   *Request;
+  IPMI_RESPONSE  *Response;
+  UINT8          ResponseSize;
+  EFI_STATUS     Status;
+
+  ASSERT (CommandDataSize <= MOCK_BASE_IPMI_BUFFER_SIZE - sizeof (IPMI_COMMAND));

```

Comment:
The expectation here is that the Mock lib will only be used with debug builds, so this assert is the only guarantee we need that the `CommandDataSize` is not too large?

Reply:
Correct, I don't expect the mock lib to be used in any production scenario, especially the base lib version, so this is just some internal checking.